src/Expectation.php

```diff
@@ -324,6 +325,39 @@ public function when(callable|bool $condition, callable $callback): self
         return $this;
     }
 
+    /**
+     * Asserts that any one of the given tests pass with the given expectation target.
+     *
+     * @param  (\Closure(self<TValue>): (mixed|void))  ...$tests
+     * @return self<TValue>
+     *
+     * @throws AssertionFailedError Rethrows first caught exception on failure
+     */
+    public function toPassAny(\Closure ...$tests): Expectation
```

Comment:
I'd like to recommend `toBeOneOf()`, and to allow non-closure values that will be converted to `toBe()` assertions. This allows for:

```php
expect($foo)->toBeOneOf(1, 2, 3);
```

Here's the code:

```php
expect()->extend('toBeOneOf', extend: function (mixed ...$potentials): Expectation {
    $firstException = null;
    if ($potentials === []) {
        return $this;
    }

    foreach ($potentials as $potential) {
        if (!($potential instanceof \Closure)) {
            $potential = fn ($expectation) => $expectation->toBe($potential);
        }

        try {
            $potential(new Expectation($this->value));

            return $this;
        } catch (AssertionFailedError $e) {
            $firstException ??= $e;
        }
    }

    throw $firstException;
});
```

Reply:
To me "toBeOneOf" both implies an exact count and hides the fact that any expectation could be provided. 

I also don't like falling back to a default expectation because in my opinion it makes the behavior confusing when your array might contain values that are closures. Id prefer an additional `->toBeAny(...)` that does something like `returb $this->toPassAny(array_map(fn($i) => fn($e) => $e->toBe($i), $items));`

Reply:
toBeAny is good, or toBeAnyOf().

You shouldn't be passing an array, it's variadic. Your example above should spread the array as args (`… array_map(…)`) or it won't work with any of the code…

Reply:
I actually like the idea of "toBeOneOf" as you say the name implies, an xor type deal. I wrote it and then ditched it but it sounds more useful with that name

Reply:
Yeah you'd need to spread it. Not the easiest code snippet to write on mobile 🙂